resources/assets/js/components/pages/admin/categories/editor/Descriptors.vue

```diff
@@ -103,10 +104,24 @@
                     icon: ""
                 });
             },
-            parseAllowedValues (descriptor) {
-                return descriptor.allowed_values.split(",").map((value) => {
+            splitTrimJoin (string, splitOn, joinWith) {
+                return string.split(splitOn).map((value) => {
                     return value.trim();
-                }).join("|");
+                }).join(joinWith);
+            },
+            // method called by parseAllowedValues when parsing a "select" descriptor
+            parseAllowedSelectValues (descriptor) {
+                return this.splitTrimJoin(descriptor.allowed_values, ",", "|");
+            },
+            parseAllowedMultiSelectValues (descriptor) {
+                return this.parseAllowedSelectValues(descriptor);
+            },
+            parseAllowedNumberValues (descriptor) {
+                return this.splitTrimJoin(descriptor.allowed_values, "-", "-");
+            },
+            parseAllowedValues (descriptor) {
+                let parser = 'parseAllowed' + window.capitalize(descriptor.value_type) + 'Values';
+                return this[parser] ? this[parser](descriptor) : descriptor.allowed_values;
```

Comment:
"magic" method that calls the proper parser for the given descriptors' value type.